src/test_mailroom.py

```diff
@@ -0,0 +1,148 @@
+        'Name{}|Total\n'.format(' ' * 20) +
+        '{}|{}\n'.format('-' * 24, '-' * 10) +
+        'David{}|{}\n'.format(' ' * 19, 50) +
+        'Steven{}|{}\n'.format(' ' * 18, 60) +
+        'Zach{}|{}\n'.format(' ' * 20, 70) +
+        '{}|{}\n'.format('-' * 24, '-' * 10)
+    )
+]
+
+SAVE_DONOR_DICT = [
+    ({'Steven': [1, 2, 3]}, 'test_save_donor.txt', 'Data saved'),
+    ({}, '#%&$/', 'Could not save to the donor.txt file.')
+]
+
+
+@pytest.mark.parametrize('donor, amount, template, message', DONOR_THANKYOU)
+def test_generate_thankyou(donor, amount, template, message):
+    """Test generate_thankyou func to output correct thankyou message based on
+    donor's name and donation amount."""
+    assert m.generate_thankyou(donor, amount, template) == message
+
+
+@pytest.mark.parametrize('donor_file_name, donor_dict', READ_DONOR_FILE)
+def test_read_donors(donor_file_name, donor_dict):
+    """Test read_donors func to read from donor.txt correctly"""
+    assert m.read_donors(donor_file_name) == donor_dict
+
+
+@pytest.mark.parametrize('user_input, func', INPUT_FUNCS)
+def test_handle_input(user_input, func):
+    """Test handle_input func if return correct func to be called at each valid
+    user input."""
+    assert m.handle_input(user_input) == func
+
+
+def test_welcome_message():
+    """Test welcone_message func to output correct info and instructions."""
+    assert m.welcome_message() == ('Welcome to the donor database.\n'
+                                   'Enter 1 to send thank you note\n'
+                                   'Enter 2 to view report\n'
+                                   'Enter 3 to exit\n')
+
+
+@pytest.mark.parametrize('donor_list, table', OUTPUT_TABLE)
+def test_build_report_table(donor_list, table):
+    """Test build_report_table func to output pretty table"""
+    assert m.build_report_table(donor_list) == table
+
+
+@pytest.mark.parametrize(
+    'donor_dict, donor_file_name, save_message', SAVE_DONOR_DICT
+)
+def test_save_donors(donor_dict, donor_file_name, save_message):
+    """Test save_donors func to make sure it save data when exit"""
+    assert m.save_donors(donor_dict, donor_file_name) == save_message
+
+
+@pytest.mark.parametrize('donor_dictionary, result', DONOR_DICT)
+def test_donor_list_by_total(donor_dictionary, result):
+    """Test donor_list_by_total func to make sure it returns a sorted list of
+    donor by total donation"""
+    assert m.donor_list_by_total(donor_dictionary) == result
+
+
+@pytest.mark.parametrize('donor_amt, result', DONATION_AMT)
+def test_validate_donation(donor_amt, result):
+    """Test validate_donation func to make sure it validates user input
+    correctly"""
+    assert m.validate_donation(donor_amt) == result
+
+
+@pytest.mark.parametrize('donor_amt, result', FORMAT_AMT)
+def test_format_amount(donor_amt, result):
+    """Test format_amount func to make sure it format dollar amount"""
+    assert m.format_amount(donor_amt) == result
```

Comment:
I'd love to see you augment these tests with some validation that things fail the way you want them to when unexpected stuff happens.